### PR TITLE
F1 help for `#if` and `#else` refers to preprocessor symbols, not `if_CSharpKeyword` and `else_CSharpkeyword`

### PR DESCRIPTION
**Pull request by BillWagner**
The F1 service should produce `#if` instead of `if_CSharpKeyword` when the cursor is on the `#if` symbol.

Fixes dotnet/docs#35715

Add tests for the F1 service on `#if` and related tokens. Then, when checking for keywords, make sure that `if` and `else` have the proper statement token as the parent token. Otherwise, that text is part of the preprocessor token.

### CODE REVIEW

**Comment by BillWagner** [comment]
ping @CyrusNajmabadi

**Comment by BillWagner** [comment]
Commit notes for reviewers:

1. **First commit**: Add other tests for the other preprocessor symbols related to conditional compilation. The `#if` and `#else` fail, because the model finds child nodes for `if` and `else`, respectively. The tests for `#elif` and `#endif` pass, because the model does not find child nodes for those tokens.
2. **Second commit**: ~~Make failing tests pass. Because the `#if` and `#else` preprocessor tokens contain the text for keywords (`if` and `else`), the `TryGetTextForKeyword` would lock to those keywords before the `TryGetTextForPreProcessor` looked for the preprocessor token. By switching the order of those two tests, the longer preprocessor tokens are found first.~~
3. **Third commit:**: Revert second commit, because that broke several tests for "default" text that was being interpreted as "#default". Instead, check the parent of the node when "if" and "else" are found. If the parent isn't the `IfStatement` or `ElseClause`, respectively, don't consider the token a keyword. The later test for a preprocessor token will find the correct node.

**Review by genlu — APPROVED** [review]
:shipit:

**Comment by CyrusNajmabadi** [comment]
looking.

**Comment by CyrusNajmabadi** [comment]
Let me take a stab at this.

File: src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs [hunk 1]
```diff
@@ -97,8 +97,8 @@ public override async Task<string> GetHelpTermAsync(Document document, TextSpan
             if (TryGetTextForSpecialCharacters(token, out var text) ||
                 TryGetTextForContextualKeyword(token, out text) ||
                 TryGetTextForCombinationKeyword(token, out text) ||
-                TryGetTextForKeyword(token, out text) ||
                 TryGetTextForPreProcessor(token, out text) ||
+                TryGetTextForKeyword(token, out text) ||
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs:101]: have pp checks come first so we defer to the pp interepretation if inside a directive.

File: src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs [hunk 2]
```diff
@@ -339,16 +339,28 @@ private static bool TryGetTextForPreProcessor(SyntaxToken token, [NotNullWhen(tr
...
+            if (directive != null)
             {
-                text = "#" + token.Text;
-                return true;
-            }
+                if (token.IsKind(SyntaxKind.DefaultKeyword) && token.Parent is LineDirectiveTriviaSyntax)
+                {
+                    text = Keyword("defaultline");
+                    return true;
+                }
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs:351]: moved from the keyword handling case to the directive handling case.

**Pull request merged**